Mathlib/Data/Set/Intervals/UnorderedInterval.lean

```diff
@@ -0,0 +1,390 @@
+
+end Lattice
+
+-- Porting note: fix scoped notation
+-- open Interval
+
+section DistribLattice
+
+variable [DistribLattice α] {a a₁ a₂ b b₁ b₂ c x : α}
+
+theorem eq_of_mem_interval_of_mem_interval (ha : a ∈ [[b, c]]) (hb : b ∈ [[a, c]]) : a = b :=
+  eq_of_inf_eq_sup_eq (inf_congr_right ha.1 hb.1) <| sup_congr_right ha.2 hb.2
+#align set.eq_of_mem_interval_of_mem_interval Set.eq_of_mem_interval_of_mem_interval
+
+theorem eq_of_mem_interval_of_mem_interval' : b ∈ [[a, c]] → c ∈ [[a, b]] → b = c := by
+  simpa only [interval_swap a] using eq_of_mem_interval_of_mem_interval
+#align set.eq_of_mem_interval_of_mem_interval' Set.eq_of_mem_interval_of_mem_interval'
+
+theorem interval_injective_right (a : α) : Injective fun b => interval b a := fun b c h => by
+  rw [ext_iff] at h
+  exact eq_of_mem_interval_of_mem_interval ((h _).1 left_mem_interval) ((h _).2 left_mem_interval)
+#align set.interval_injective_right Set.interval_injective_right
+
+theorem interval_injective_left (a : α) : Injective (interval a) := by
+  simpa only [interval_swap] using interval_injective_right a
+#align set.interval_injective_left Set.interval_injective_left
+
+end DistribLattice
+
+section LinearOrder
+
+variable [LinearOrder α] [LinearOrder β] {f : α → β} {s : Set α} {a a₁ a₂ b b₁ b₂ c x : α}
+
+theorem Icc_min_max : Icc (min a b) (max a b) = [[a, b]] :=
+  rfl
+#align set.Icc_min_max Set.Icc_min_max
+
+theorem interval_of_not_le (h : ¬a ≤ b) : [[a, b]] = Icc b a :=
+  interval_of_gt <| lt_of_not_ge h
+#align set.interval_of_not_le Set.interval_of_not_le
+
+theorem interval_of_not_ge (h : ¬b ≤ a) : [[a, b]] = Icc a b :=
+  interval_of_lt <| lt_of_not_ge h
+#align set.interval_of_not_ge Set.interval_of_not_ge
+
+theorem interval_eq_union : [[a, b]] = Icc a b ∪ Icc b a := by rw [Icc_union_Icc', max_comm] <;> rfl
+#align set.interval_eq_union Set.interval_eq_union
+
+theorem mem_interval : a ∈ [[b, c]] ↔ b ≤ a ∧ a ≤ c ∨ c ≤ a ∧ a ≤ b := by simp [interval_eq_union]
+#align set.mem_interval Set.mem_interval
+
+theorem not_mem_interval_of_lt (ha : c < a) (hb : c < b) : c ∉ [[a, b]] :=
+  not_mem_Icc_of_lt <| lt_min_iff.mpr ⟨ha, hb⟩
+#align set.not_mem_interval_of_lt Set.not_mem_interval_of_lt
+
+theorem not_mem_interval_of_gt (ha : a < c) (hb : b < c) : c ∉ [[a, b]] :=
+  not_mem_Icc_of_gt <| max_lt_iff.mpr ⟨ha, hb⟩
+#align set.not_mem_interval_of_gt Set.not_mem_interval_of_gt
+
+theorem interval_subset_interval_iff_le :
+    [[a₁, b₁]] ⊆ [[a₂, b₂]] ↔ min a₂ b₂ ≤ min a₁ b₁ ∧ max a₁ b₁ ≤ max a₂ b₂ :=
+  interval_subset_interval_iff_le'
+#align set.interval_subset_interval_iff_le Set.interval_subset_interval_iff_le
+
+/-- A sort of triangle inequality. -/
+theorem interval_subset_interval_union_interval : [[a, c]] ⊆ [[a, b]] ∪ [[b, c]] := fun x => by
+  simp only [mem_interval, mem_union]
+  cases' le_total a c with h1 h1 <;>
+  cases' le_total x b with h2 h2 <;>
+  -- Porting note: restore `tauto`
```

Comment:
```suggestion
```
If `aesop` works, that's fine.

Reply:
In some cases Aesop actually didn’t work where tauto used to, and I’ve written up a proof manually. Where Aesop did work, I’ve also sometimes replaced it with the generated proof